MyPenelopF/src/main/java/classes/Group.java

```diff
@@ -1,5 +1,89 @@
 package classes;
 
+import java.util.ArrayList;
+
 public class Group {
+	
+	private Integer id;
+	
+	private String name;
+	
+	private ArrayList<Contact> users;
+	
+	private ArrayList<Project> projects;
+	
+	// Database auto-increment id simulation, must be enhanced by a "retrieveLastId" method
+	static private int autoincrement = 0;
+	static private int increment() {
+		return ++Group.autoincrement;
+	}
+	
+	//default constructor
+	public Group() {}
+	
+	//Surcharged minimal constructor
+	public Group(String name) {
+		this.setId(Group.increment());
```

Comment:
this.id = Group.increment();
this.name = name;

ça évite un appel de méthode. Appel méthode + affectation > affectation en terme de perfs